datafusion/core/src/datasource/stream.rs

```diff
@@ -103,19 +105,29 @@ impl FromStr for StreamEncoding {
     }
 }
 
-/// The configuration for a [`StreamTable`]
+pub trait StreamSource: std::fmt::Debug + Send + Sync {
```

Comment:
Perhaps some documentation about what this trait is meant to represent would help